hamilton/plugins/dlt_extensions.py

```diff
@@ -0,0 +1,128 @@
+import dataclasses
+from typing import Any, Collection, Dict, Iterable, Literal, Optional, Sequence, Tuple, Type
+
+import dlt
+import pandas as pd
+from dlt.common.destination import Destination, TDestinationReferenceArg  # noqa: F401
+from dlt.common.schema import Schema, TColumnSchema
+
+# importing TDestinationReferenceArg fails if Destination isn't imported
+from dlt.extract.resource import DltResource
+
+from hamilton import registry
+from hamilton.io import utils
+from hamilton.io.data_adapters import DataLoader, DataSaver
+
+DATAFRAME_TYPES = [Iterable, pd.DataFrame]
+
+# TODO add types for other Dataframe libraries
+try:
+    import pyarrow as pa
+
+    DATAFRAME_TYPES.extend([pa.Table, pa.RecordBatch])
+except ModuleNotFoundError:
```

Comment:
So this works, the problem is that the error message will be confusing... For now, however, I think we can just document it well.

Reply:
Why would we want to throw an error? You mean if a node has annotions for `pa.Table`, but pyarrow is not installed and therefore finds no registered materializers?

Reply:
Yeah, exactly. Its just a confusing case, but let's not worry about it (maybe add to docs).